### PR TITLE
AuditReader::find and AuditReader::findRevision should not throw exception and return null instead

### LINKED ISSUE #1085: [NEXT_MAJOR] AuditReader::find and AuditReader::findRevision should not throw exception and return null instead.
## Feature Request

Related to https://github.com/sonata-project/SonataAdminBundle/pull/6261#discussion_r467395214

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

I am targeting this branch, because BC-break.

Closes #1085.

### CODE REVIEW

**Review by OskarStark — APPROVED** [review]

**Review by dmaicher — APPROVED** [review]
Should this be covered by tests?

**Comment by VincentLanglet** [comment]
> Should this be covered by tests?

It can be, but I will first fix the master branch before finishing this PR

**Review by phansys — APPROVED** [review]

**Comment by VincentLanglet** [comment]
See https://github.com/sonata-project/EntityAuditBundle/pull/373 a lot of exceptions can be thrown

File: src/Model/AuditReader.php [hunk 1]
```diff
@@ -30,7 +30,11 @@ public function __construct(SimpleThingsAuditReader $auditReader)
 
     public function find(string $className, $id, $revisionId): ?object
     {
-        return $this->auditReader->find($className, $id, $revisionId);
+        try {
+            return $this->auditReader->find($className, $id, $revisionId);
+        } catch (\Exception $exception) {
```
Comment (phansys) [inline src/Model/AuditReader.php:35]: If we haven't specific domain exceptions for these situations, shouldn't we catch throwable?

Comment (VincentLanglet) [reply src/Model/AuditReader.php]: Done

**Review by phansys — APPROVED** [review]

**Pull request merged**